masonry/src/widget/window_handle.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024 the Xilem Authors and the Druid Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! A titlebar area widget.
+
+use accesskit::{NodeBuilder, Role};
+use dpi::LogicalPosition;
+use smallvec::{smallvec, SmallVec};
+use tracing::{trace_span, Span};
+use vello::kurbo::Point;
+use vello::Scene;
+
+use crate::event::PointerButton;
+use crate::widget::{WidgetMut, WidgetPod};
+use crate::{
+    AccessCtx, AccessEvent, BoxConstraints, EventCtx, LayoutCtx, LifeCycleCtx, PaintCtx,
+    PointerEvent, Size, StatusChange, TextEvent, Widget, WidgetId,
+};
+
+// TODO: Maybe make this configurable somehow
+const DRAG_THRESHOLD: f64 = 8.0;
+
+/// A titlebar area widget.
+///
+/// An area that can be dragged to move the window.
+pub struct WindowHandle<W: Widget> {
+    child: Option<WidgetPod<W>>,
+    last_pos: Option<LogicalPosition<f64>>,
+}
+
+// --- MARK: BUILDERS ---
+impl<W: Widget> WindowHandle<W> {
+    pub fn new(child: W) -> Self {
```

Comment:
So the idea is that you would put a `Flex` (or w/e) containing your titlebar buttons/icon (/maybe window menu) in here, and the non-interactive areas would be treated as a draggable area?

Reply:
Basically, yes.